IDE/Espressif/ESP-IDF/user_settings.h

```diff
@@ -21,20 +21,18 @@
 #undef WOLFSSL_ESPIDF
 #undef WOLFSSL_ESPWROOM32
 #undef WOLFSSL_ESPWROOM32SE
-#undef WOLFSSL_ESPWROOM32
-#undef WOLFSSL_ESP8266
 
 #define WOLFSSL_ESPIDF
 
 /*
- * choose ONE of these Espressif chips to define:
+ * choose ONE of these Espressif chips to define for HW acceleration or
+ * leave all undefined for a non-accelerated build for other chips:
  *
  * WOLFSSL_ESPWROOM32
  * WOLFSSL_ESPWROOM32SE
- * WOLFSSL_ESP8266
  */
 
-#define WOLFSSL_ESPWROOM32
+/* #define WOLFSSL_ESPWROOM32 */
```

Comment:
Ok, I'll look into adding some #ifdefs to enable it on the appropriate targets.

Reply:
Please help me clarify a few things here if I read the code correctly:
- `WOLFSSL_ESPWROOM32` has nothing to do with the WROOM module, it's the internal crypto accelerator of the ESP32 chip. and should essentially be `CONFIG_IDF_TARGET_ESP32`. The other targets have different crypto accelerators.
- the WROOM32-SE module has the ATECC608A in addition. `WOLFSSL_ESPWROOM32SE` enables the same features as `WOLFSSL_ESPWROOM32` and adds the WOLFSSL_ATECC508A #define (although the code has 608A define, too).

Reply:
Yes, you understand correctly. Unfortunately that's true, and not very intuitive, I know. That was created when there was only the ESP32WROOM chip. At some point that needs to be cleaned up. 

I agree the `WOLFSSL_ESPWROOM32`  and `WOLFSSL_ESPWROOM32SE` should probably be completely removed and replaced with something more appropriate.

The ESP32-S3 on the Xtensa architecture has very similar crypto accelerators. As you noted: the RISC-V chipsets are very different. I have a WIP for some HW acceleration on the -C3 and -C6.

Oh, and yes: the `SE` uses the ATECC508A. I'm not certain if it is 100% compatible with the ATECC608A.

Reply:
According to the [datasheet](https://www.espressif.com/sites/default/files/documentation/esp32-wroom-32se_datasheet_en.pdf), the SE has a 608A. Do you maybe have the hardware to test if changing the define to 608 still works?

Otherwise how about this:
- Remove `WROOM32` from the following macros, they all refer to the internal crypto in the ESP32:  
  - `NO_ESP32WROOM32_CRYPT`
  - `WOLFSSL_ESP32WROOM32_CRYPT`
  - `NO_WOLFSSL_ESP32WROOM32_CRYPT_RSA_PRI`
  - `NO_WOLFSSL_ESP32WROOM32_CRYPT_HASH`
  - `NO_WOLFSSL_ESP32WROOM32_CRYPT_AES`
  - `WOLFSSL_ESP32WROOM32_CRYPT_DEBUG`
  - `WOLFSSL_USE_ESP32WROOM32_CRYPT_HASH_HW`
  - `WOLFSSL_ESP32WROOM32_CRYPT_RSA_PRI`
- Replace `WOLFSSL_ESPWROOM32` with `CONFIG_IDF_TARGET_ESP32`. It is only used in settings.h/user_settings.h to set other macros. HW crypto can still be disabled from user_settings.h with the NO_ macros.
- Leave `WOLFSSL_ESPWROOM32SE` as is (maybe change 508A to 608A) to enable the Atmel chip from user_settings.h, by default disabled.
- If S3 hardware crypto is merged and shares a significant amount of code with the original ESP32, maybe also set some `WOLFSSL_ESP32_XTENSA_CRYPT` based on `CONFIG_IDF_TARGET` in settings.h for both to mark the shared code.

Reply:
My mistake on the SE, I thought it was the '608A. I don't have either one to test.

I agree on others: I like the idea of removing `WROOM32`. Good to keep `WOLFSSL_ESPWROOM32SE` as it is a very specific chipset. At some point, I'd like to have 508A/608A support on other chipsets that may have a manually added device.

Reply:
Done, PTAL.